Projektskabelon-master/src/main/java/FunctionLayer/OverlayMaterialCalculator.java

```diff
@@ -231,18 +239,21 @@ public  static Material screwForOverlayOneWall ( Wall wall, String overlayName)
         screwFyr.setName("BASIC SKRUE 5,0X40MM");
         screwFyr.setSize(screwFyrQuantity); //for the door - always the same size of the door
         screwFyr.setComment("Til montering af vertical framing-dør");
+        screwFyr.setPrice(LogicFacade.getPrice(screwFyr.getId()));
         doorMaterials.add(screwFyr);
 
         Material screwFyrDoor = new Material();
         screwFyrDoor.setName("BASIC SKRUE 5,0X40MM");
         screwFyrDoor.setSize(SCREWFORDOORELEMEN*DOORMETALELEMENS); //for the door - always the same size of the door
         screwFyrDoor.setComment("Til montering af vertical framing-dør");
+        screwFyrDoor.setPrice(LogicFacade.getPrice(screwFyrDoor.getId()));
         doorMaterials.add(screwFyrDoor);
 
         Material screwSpaer = new Material();
         screwSpaer.setName("5X80 MM RUST FRI SKRUER");
         screwSpaer.setComment("til montering af horizontal framing-dør");
         screwSpaer.setSize(screwSpaerQuantity); //for the door - always the same sioze of the door
+        screwSpaer.setPrice(LogicFacade.getPrice(screwSpaer.getId()));
```

Comment:
If you set the price here, it will be calculated for the needed size, not for the available one. I calculate de available sizes first in function layer